app/controllers/file.js

```diff
@@ -17,6 +17,39 @@ const FileController = function(url, header) {
 
 util.inherits(FileController, EventEmitter);
 
+/* Download file and save to disk. */
+FileController.prototype.downloadFile = function() {
```

Comment:
@donnapep This original downloadFile function caused problems when trying to reuse it because it includes the file events and also streaming the response. In the scenario where the file exists and it's stale, we need a function that just makes a request with optional headers and handles 200 to download the file or 304 to just save headers.